private_gpt/components/embedding/embedding_component.py

```diff
@@ -7,7 +7,7 @@
 from private_gpt.settings.settings import Settings
 
 logger = logging.getLogger(__name__)
-
+import torch
```

Comment:
I'd move this to the `try` block within "huggingface" case. There is no "torch" general dependency declared in pyproject.toml, so this could break the whole execution for people not using huggingface. Actually, we may need to add torch to `embeddings-huggingface = ["llama-index-embeddings-huggingface"]` as
```
# Optional Huggingface related dependency
torch = {version = "^2.2.1", optional = true}

embeddings-huggingface = ["torch", "llama-index-embeddings-huggingface"] 
```
in pyproject.toml. 

I think huggingface package from llamaindex already depends on torch, but given we are now importing it explicitly we should also depende on it.